code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm

```diff
@@ -0,0 +1,188 @@
+	name = "Drain"
+	action_icon_state = "drain"
+	mechanics_text = "Hold a marine for 2s while draining 2% of their blood."
+	ability_name = "drain"
+	cooldown_timer = 15 SECONDS
+	plasma_cost = 50
+
+/datum/action/xeno_action/activable/drain/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	var/mob/living/carbon/xenomorph/gorger/X = owner
+	if(!.)
+		return
+	if(!iscarbon(A))
+		return FALSE
+	if(!A.can_sting())
+		to_chat(owner, span_warning("This won't do!"))
+		return FALSE
+	if(X.blood_bank >= 100)
+		to_chat(owner, span_warning("No need, we feel sated for now..."))
+		return FALSE
+	if(get_dist(owner, A) > 1)
+		to_chat(owner, span_warning("It is outside of our reach! We need to be closer!"))
+		return FALSE
+	return TRUE
+
+/datum/action/xeno_action/activable/drain/use_ability(mob/living/carbon/A)
+	var/mob/living/carbon/xenomorph/gorger/X = owner
+	X.do_attack_animation(A, ATTACK_EFFECT_GRAB)
+	X.emote("roar")
+	A.drop_all_held_items()
+	X.visible_message(A, span_danger("The [X] grabs [A]!"))
+	A.SetImmobilized(2 SECONDS)
+	if(!do_after(X, 2 SECONDS, TRUE, A, BUSY_ICON_HOSTILE, ignore_turf_checks = FALSE))
+		to_chat(X, span_warning("Our meal is interrupted!"))
+		return
+	X.do_attack_animation(A, ATTACK_EFFECT_REDSTAB)
+	A.emote("scream")
+	A.drop_all_held_items()
+	X.visible_message(A, span_danger("The [X] stabs its tail into [A]!"))
+	A.apply_damage(damage = 20, damagetype = BRUTE, def_zone = BODY_ZONE_HEAD, blocked = 0, sharp = TRUE, edge = FALSE, updating_health = TRUE)
+	playsound(A, "alien_claw_flesh", 25, TRUE)
+	A.blood_volume -= 10
+	X.blood_bank += min(10, 100 - X.blood_bank)
+	to_chat(X, span_notice("Blood bank: [X.blood_bank]%"))
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Rejuvenate/Transfusion
+// ***************************************
+/datum/action/xeno_action/activable/rejuvenate
+	name = "Rejuvenate/Transfusion"
+	action_icon_state = "Rejuvenate"
+	mechanics_text = "When used on self, drains 10 blood and restores 40% health over 5s. When used on another xenomorph, restores 30% of their health after 2s. When used on a dead human, you grab them and heal by draining their blood."
+	ability_name = "rejuvenate"
+	cooldown_timer = 2 SECONDS
+	plasma_cost = 0
+	use_state_flags = XACT_TARGET_SELF
+
+/datum/action/xeno_action/activable/give_action(mob/living/L)
+	. = ..()
+
+/datum/action/xeno_action/activable/rejuvenate/can_use_ability(atom/target, silent = FALSE, override_flags) //it is set up to only return true on specific xeno or human targets
+	. = ..()
+	if(!.)
+		return
+	var/mob/living/carbon/xenomorph/gorger/X = owner
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		if(H.stat != DEAD)
+			to_chat(X, span_notice("We can only use still, dead hosts to heal."))
+			return FALSE
+		else if(get_dist(X, H) > 1)
+			to_chat(X, span_notice("We need to be next to our meal."))
+			return FALSE
+		return TRUE
+	if(isxeno(target))
```

Comment:
probably could flip this to make it nicer.

Reply:
Do you mean placing the isxeno check above the ishuman check? I think the proc will get into the ishuman check more often, which is why I think it should be above.

Reply:
you coul invert the checks (ie !isxeno) to early return earler